CBinding/CMake/CMakeProject.cs

```diff
@@ -0,0 +1,349 @@
+﻿//
+// CMakeProject.cs
+//
+// Author:
+//       Elsayed Awdallah <comando4ever@gmail.com>
+//
+// Copyright (c) 2015 Xamarin Inc. (http://xamarin.com)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.IO;
+using System.Text;
+
+using MonoDevelop.Core;
+using MonoDevelop.Core.Execution;
+using MonoDevelop.Projects;
+
+namespace CBinding {
+	public class CMakeProject : SolutionItem {
+		FilePath file;
+		string name;
+		string outputDirectory = "./bin";
+		CMakeFileFormat fileFormat;
+		
+		Stream executeCommand (string command, string args, string workingDir, ProgressMonitor monitor)
+		{
+			MemoryStream stream = new MemoryStream ();
+			StreamWriter streamWriter = new StreamWriter (stream);
+			string path = Path.Combine (file.ParentDirectory.ToString (), workingDir);
+			ProcessWrapper p = Runtime.ProcessService.StartProcess (command, args, path, monitor.Log, streamWriter, null);
+			p.WaitForExit ();
+			streamWriter.Flush ();
+			stream.Position = 0;
+			return stream;
+		}
+		
+		bool checkCMake ()
+		{
+			try {
+				ProcessWrapper p = Runtime.ProcessService.StartProcess ("cmake", "--version", null, null);
+				p.WaitForOutput ();
+				return true;
+			} catch {
+				return false;
+			}
+		}
+		
+		Tuple<int, string> getFileAndLine (string line, string separator)
+		{
+			int lineNumber = 0;
+			string fileName = "";
+			string s = line.Split (new String[] {separator}, StringSplitOptions.RemoveEmptyEntries) [1].Trim ();
+			string[] args = s.Split(':');
+			if (args [0].Length > 0) fileName = args [0];
+			if (args.Length > 1 && args [1].Length > 0) {
+				if(args [1].Contains ("("))
+					int.TryParse (args [1].Split ('(') [0], out lineNumber);
+				else
+					int.TryParse (args [1], out lineNumber);
+			}
+			
+			return new Tuple<int, string> (lineNumber, fileName);
```

Comment:
FWIW Tuple.Create is a bit neater as it uses type inference (ctors can't use type inference)

    return Tuple.Create (lineNumber, fileName);